crates/rpc-client/src/client.rs

```diff
@@ -163,6 +163,8 @@ pub struct RpcClientInner<T> {
     pub(crate) id: AtomicU64,
     /// The poll interval for the client in milliseconds.
     pub(crate) poll_interval: AtomicU64,
+    /// `true` if the poll interval is custom.
+    pub(crate) is_custom_poll_interval: AtomicBool,
```

Comment:
don't really like the `custom` part because we now also set this based on the chainid, which isn't really custom imo.

but can't come up with better alternatives rn.